### PR TITLE
Hide top ad banners in adfree mode

### PR DESCRIPTION
**Pull request by jbreckmckye**
- In adfree mode, hide the top banner with CSS before they even render (to avoid any flashing). We append a `:not(.is-adfree)` to the selectors which turn on slots for different breakpoints. IE8 doesn't get adverts, so its lack of :not support isn't a problem.
- In DFP.js, strip the ad slot from the DOM so that DFP doesn't make any redundant ad calls (which would waste network resources and skew our view measurements)

### CODE REVIEW

File: static/src/stylesheets/module/_adslot.scss [hunk 1]
```diff
@@ -120,14 +120,14 @@
 }
 .top-banner-ad-container--above-nav {
     @include mq(tablet) {
-        .js-on & {
+        .js-on:not(.is-adfree) & {
```
Comment (Calanthe) [inline static/src/stylesheets/module/_adslot.scss]: Where do you apply that `is-adfree` class?

Comment (jbreckmckye) [inline static/src/stylesheets/module/_adslot.scss]: It's from the preflight scripts. See https://github.com/guardian/frontend/pull/10909

**Comment by Calanthe** [comment]
It seems fine :+1:

**Comment by paperboyo** [comment]
Oh, so we do have a way to fix huge-white-space-above-nav-in-ff-with-adblock? (https://github.com/guardian/frontend/pull/9789#issuecomment-133370552)

**Pull request merged**